impl/src/main/java/io/kroxylicious/testing/kafka/common/KeystoreManager.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.bouncycastle.util.IPAddress;
+
+import io.netty.pkitesting.CertificateBuilder;
+import io.netty.pkitesting.X509Bundle;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class KeystoreManager {
+    private String password;
+
+    /**
+     * Creates a CertificateBuilder with the appropriate default values for kroxylicious test usage.
+     * @param distinguishedName the distinguished name. See {@link KeystoreManager#buildDistinguishedName(String, String, String, String, String, String, String)}
+     *                          for generating a distinguished name
+     * @return  The partially populated certificate builder.
+     */
+    public CertificateBuilder newCertificateBuilder(String distinguishedName) {
+        return newCertificateBuilder(distinguishedName, new ArrayList<>());
+    }
+
+    /**
+     * Creates a CertificateBuilder with the appropriate default values for kroxylicious test usage.
+     * @param distinguishedName the distinguished name. See {@link KeystoreManager#buildDistinguishedName(String, String, String, String, String, String, String)}
+     *                          for generating a distinguished name
+     * @param sanNames the names for SAN: They can be DNS names and/or IP Addresses
+     * @return  The partially populated certificate builder.
+     */
+    public CertificateBuilder newCertificateBuilder(String distinguishedName, @NonNull List<String> sanNames) {
```

Comment:
Hmm I understand this is how we want to use it now but somewhat defeats the point of returning the certificate builder. 

I think we merge this signature now to make progress. I'm tempted to say we should mark the the SAN names as optional in javadoc however.

Reply:
Is it worth it? Why not just let users add them directly to the builder?

I'm happy to add the method if there is value I'm just not seeing it right now